tests/goth_tests/test_mid_agreement_payments/test_mid_agreement_payments.py

```diff
@@ -1,43 +1,52 @@
 """A goth test scenario for mid-agreement payments."""
-from functools import partial
+from goth.assertions import EventStream
+from goth.configuration import Override, load_yaml
+from goth.runner import Runner
+from goth.runner.log import configure_logging
+from goth.runner.probe import RequestorProbe
 import logging
 import os
 from pathlib import Path
-import re
-from typing import List
-
 import pytest
+import re
+from typing import List, Optional
 
-from goth.configuration import load_yaml, Override
-from goth.runner.log import configure_logging
-from goth.runner import Runner
-from goth.runner.probe import RequestorProbe
 from yapapi.strategy import DEBIT_NOTE_INTERVAL_GRACE_PERIOD
 
+from . import requestor_agent
 
 logger = logging.getLogger("goth.test.mid_agreement_payments")
 
 
-async def check_debit_note_freq(events):
-    freq = None
-    expected_note_num = 1
-    total_time = 0
-    note_num = None
-    async for line in events:
-        if freq is None:
-            m = re.search(r"Debit notes interval: ([0-9]+)", line)
-            if m:
-                freq = int(m.group(1))
-        m = re.search(r"Debit notes for activity.* ([0-9]+) notes/([0-9]+)", line)
-        if m and "Payable" not in line:
-            note_num = int(m.group(1))
-            total_time = int(m.group(2))
-            assert note_num == expected_note_num, "Unexpected debit note number"
-            expected_note_num += 1
-            assert freq is not None, "Expected debit note frequency message before a debit note"
-            assert total_time + DEBIT_NOTE_INTERVAL_GRACE_PERIOD > note_num * freq, "Too many notes"
-    assert note_num is not None and note_num >= 2, "Expected at least two debit notes"
-    return f"{note_num} debit notes processed"
+async def assert_debit_note_freq(events: EventStream) -> str:
+    expected_note_num: int = 1
+    received_freq: Optional[int] = None
+    received_note_num: int = 0
+    async for log_line in events:
+        if received_freq is None:
+            freq_match = re.search(r"Debit notes interval: (?P<freq>[0-9]+)", log_line)
+            if freq_match:
+                received_freq = int(freq_match.groupdict()["freq"])
```

Comment:
and same here... while it's _not_ an error to use named groups, again, I think that the code gets _more_ verbose and readability _suffers_ instead of being improved... please re-think your approach to such changes...